plugins/events-backend/src/service/DefaultEventBroker.ts

```diff
@@ -14,57 +14,49 @@
  * limitations under the License.
  */
 
+import { LoggerService } from '@backstage/backend-plugin-api';
 import {
+  DefaultEventService,
   EventBroker,
   EventParams,
+  EventService,
   EventSubscriber,
 } from '@backstage/plugin-events-node';
-import { Logger } from 'winston';
 
 /**
  * In process event broker which will pass the event to all registered subscribers
  * interested in it.
  * Events will not be persisted in any form.
  *
  * @public
+ * @deprecated use `DefaultEventService` from `@backstage/plugin-events-node` instead
  */
-// TODO(pjungermann): add prom metrics? (see plugins/catalog-backend/src/util/metrics.ts, etc.)
 export class DefaultEventBroker implements EventBroker {
-  constructor(private readonly logger: Logger) {}
+  private readonly eventService: EventService;
 
-  private readonly subscribers: {
-    [topic: string]: EventSubscriber[];
-  } = {};
+  /**
+   *
+   * @param logger - logger
+   * @param eventService - replacement that gets wrapped to support not yet migrated implementations.
+   * An instance can be passed (required for a mixed mode), otherwise a new instance gets created internally.
+   * @deprecated use `DefaultEventService` directly instead
+   */
+  constructor(logger: LoggerService, eventService?: EventService) {
+    this.eventService = eventService ?? DefaultEventService.create({ logger });
```

Comment:
Defaulting the implementation doesn't really work, does it? 🤔

Reply:
@Rugvip I cannot follow you here. We use a default for `eventService` here. What do you refer to?

`DefaultEventBroker` has a different interface, but delegates to the new `DefaultEventService` internally. You could pass an instance or use a default instance.

If you want to use only the `DefaultEventBroker` instance, it's fine. (Old setup)
If you want to use old + new, you would need to pass the instance you use at other places. Events will only flow within one of them, otherwise.

E.g., event `foo` published to the `DefaultEventBroker` instance will not end up at the subscriber at a `DefaultEventService` instance if that instance was not passed as an argument when creating the broker.